src/Controller/SpaceController.php

```diff
@@ -11,15 +12,28 @@
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
-#[Route('/space')]
 class SpaceController extends AbstractController
 {
-    #[Route('/', name: 'space_index', methods: ['GET'])]
-    public function index(SpaceRepository $spaceRepository): Response
+
+    /**
+     * @Route("/", name="app_home")
+     */
+    public function index(Request $request, SpaceRepository $spaceRepository): Response
     {
-        return $this->render('space/index.html.twig', [
-            'spaces' => $spaceRepository->findAll(),
-        ]);
+
+        $form = $this->createForm(SearchType::class, null, array('method' => 'GET'));
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $location = $form->get('location')->getData();
+
+            return $this->render('space/list.html.twig', [
+                'location' => $location,
+                'spaces' => $spaceRepository->findByLocation($location)
+            ]);
+        }
+
+        return $this->renderForm('space/index.html.twig', ['form' => $form]);
```

Comment:
add $spaces and $location to the array (and handle in the template that location may be null)